Assets/Scripts/Gameplay/GameplayObjects/Character/ClientPlayerAvatarNetworkAnimator.cs

```diff
@@ -0,0 +1,70 @@
+using Unity.Netcode;
+using Unity.Netcode.Components;
+using UnityEngine;
+
+namespace Unity.BossRoom.Gameplay.GameplayObjects.Character
+{
+    /// <summary>
+    /// Component that spawns a PlayerAvatar's Avatar. It does this in two places:
+    /// 1) either inside OnNetworkSpawn() or
+    /// 2) inside NetworkAnimator's OnSynchronize method.
+    /// The latter is necessary for clients receiving initial synchronizing data, where the Animator needs to be present
+    /// and bound (Animator.Bind()) *before* the incoming animation data is applied.
+    /// </summary>
+    public class ClientPlayerAvatarNetworkAnimator : NetworkAnimator
+    {
+        [SerializeField]
+        NetworkAvatarGuidState m_NetworkAvatarGuidState;
+
+        bool m_AvatarInstantiated;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (!IsClient || m_AvatarInstantiated)
+            {
+                return;
+            }
+
+            InstantiateAvatar();
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            m_AvatarInstantiated = false;
+            var avatarGraphics = Animator.transform.GetChild(0);
+            if (avatarGraphics != null)
+            {
+                Destroy(avatarGraphics.gameObject);
+            }
+        }
+
+        protected override void OnSynchronize<T>(ref BufferSerializer<T> serializer)
```

Comment:
Nice way to do this! 👍 
I am planning on adding a prespawn and postspawn sometime in the next few weeks which would allow you to do this in the NetworkBehaviour.OnPrespawn method... but for now it is the best approach (imo).

Reply:
Awesome. Let us know when that lands, and we'll take a look! 😄